### ISSUE TITLE
false positive: use_g_autoptr_inline_cleanup

### ISSUE THREAD

**Issue by blandford** [bilelmoussaoui/gobject-lint#33]
Maybe an anti-pattern, as there are other ways to do this, but I have the following code. It wants me to use `g_autoptr(filter)` but I'm reusing the variable several times.

```C
  g_autoptr (GtkFileDialog) file_dialog = NULL;
  g_autoptr (GListStore) filters = NULL;
  GtkFileFilter *filter = NULL;

  file_dialog = gtk_file_dialog_new ();
  gtk_file_dialog_set_title (file_dialog, _("Select a crossword"));

  filters = g_list_store_new (GTK_TYPE_FILE_FILTER);

  filter = gtk_file_filter_new ();
  gtk_file_filter_set_name (filter, _("All Puzzles"));
  gtk_file_filter_add_pattern (filter, "*.ipuz");
  gtk_file_filter_add_pattern (filter, "*.puz");
  gtk_file_filter_add_pattern (filter, "*.xml");
  gtk_file_filter_add_pattern (filter, "*.jpz");
  gtk_file_filter_add_pattern (filter, "*.jpuz");
  gtk_file_filter_add_pattern (filter, "*.xd");
  g_list_store_append (filters, filter);
  gtk_file_dialog_set_default_filter (file_dialog, filter);
  g_object_unref (filter);

  filter = gtk_file_filter_new ();
  gtk_file_filter_set_name (filter, _("Ipuz Puzzles (.ipuz)"));
  gtk_file_filter_add_pattern (filter, "*.ipuz");
  g_list_store_append (filters, filter);
  g_object_unref (filter);

  filter = gtk_file_filter_new ();
  gtk_file_filter_set_name (filter, _("AcrossLite Puzzles (.puz)"));
  gtk_file_filter_add_pattern (filter, "*.puz");
  g_list_store_append (filters, filter);
  g_object_unref (filter);
  ...
```

**Comment by bilelmoussaoui** [comment]
I guess one can still keep this one as a suggestion, you can always use autoptr and drop the last g_object_unref?

**Comment by amolenaar** [comment]
One other thing: The linter recommends using `g_autoptr`, but this construct only works for GNU/clang, not MSVC. If you want portable code, you should not use this.

**Comment by bilelmoussaoui** [comment]
> One other thing: The linter recommends using `g_autoptr`, but this construct only works for GNU/clang, not MSVC. If you want portable code, you should not use this.

It is a rule, you can set it to ignore if you want to use msvc features. But majority of GObject code is likely to run on Linux only

**Comment by bilelmoussaoui** [comment]
I am leaning towards marking this one as not an issue, as you could `g_steal_pointer (&filter)` in the places you are passing the filter and that should be good as well?
